HSEARCH-4845 Use a different field for retrieving a root object id in the Lucene backend (#3489)
hibernate/hibernate-search: documentation/src/main/asciidoc/migration/index.asciidoc

[[data-format]]
== Data format and schema changes

Elasticsearch indexes created with Hibernate Search {hibernateSearchPreviousStableVersionShort}
can be read from and written to with Hibernate Search {hibernateSearchVersion}.

Reading and writing to Lucene indexes created with Hibernate Search {hibernateSearchPreviousStableVersionShort}
using Hibernate Search {hibernateSearchVersion} may lead to exceptions, since there were incompatible changes applied to internal fields.

> It is recommended to recreate your Lucene indexes and reindex your database. The easiest way to do so is to use link:{hibernateSearchDocUrl}#indexing-massindexer[the `MassIndexer`] with link:{hibernateSearchDocUrl}#indexing-massindexer-parameters-drop-and-create-schema[`dropAndCreateSchemaOnStart(true)`].

Reviewer: Actually you can be more forceful here since entity loading -- the default and probably only projection used by the vast majority of users -- requires to retrieve the identifier from the index.

```suggestion
You must recreate your Lucene indexes and reindex your database. The easiest way to do so is to use link:{hibernateSearchDocUrl}#indexing-massindexer[the `MassIndexer`] with link:{hibernateSearchDocUrl}#indexing-massindexer-parameters-drop-and-create-schema[`dropAndCreateSchemaOnStart(true)`].
```
